openmetadata-ui/src/main/resources/ui/src/pages/AddNotificationPage/AddNotificationPage.tsx

```diff
@@ -0,0 +1,374 @@
+        t('server.entity-fetch-error', { entity: t('label.config') })
+      );
+    } finally {
+      setLoadingCount((count) => count - 1);
+    }
+  };
+
+  useEffect(() => {
+    fetchFunctions();
+    if (!fqn) {
+      return;
+    }
+    fetchAlerts();
+  }, [fqn]);
+
+  const isEditMode = useMemo(() => !isEmpty(fqn), [fqn]);
+
+  const handleSave = async (data: CreateEventSubscription) => {
+    try {
+      setIsButtonLoading(true);
+
+      const destinations = data.destinations?.map((d) => ({
+        type: d.type,
+        config: d.config,
+        category: d.category,
+      }));
+
+      if (fqn && !isUndefined(alert)) {
+        const { resources, ...otherData } = data;
+
+        // Remove 'destinationType' field from the `destination` as it is used for internal UI use
+        const initialDestinations = alert.destinations.map((d) => {
+          const { destinationType, ...originalData } = d;
+
+          return originalData;
+        });
+
+        const jsonPatch = compare(
+          { ...alert, destinations: initialDestinations },
+          {
+            ...alert,
+            ...otherData,
+            filteringRules: {
+              ...alert.filteringRules,
+              resources,
+            },
+            destinations,
+          }
+        );
+
+        await updateNotificationAlert(alert.id, jsonPatch);
+      } else {
+        await createNotificationAlert({
+          ...data,
+          destinations,
+        });
+      }
+
+      showSuccessToast(
+        t(`server.${isEditMode ? 'update' : 'create'}-entity-success`, {
+          entity: t('label.alert-plural'),
+        })
+      );
+      history.push(getNotificationAlertDetailsPath(data.name));
+    } catch (error) {
+      if (
+        (error as AxiosError).response?.status === HTTP_STATUS_CODE.CONFLICT
+      ) {
+        showErrorToast(
+          t('server.entity-already-exist', {
+            entity: t('label.alert'),
+            entityPlural: t('label.alert-lowercase-plural'),
+            name: data.name,
+          })
+        );
+      } else {
+        showErrorToast(
+          error as AxiosError,
+          t(
+            `server.${
+              isEditMode ? 'entity-updating-error' : 'entity-creation-error'
+            }`,
+            {
+              entity: t('label.alert-lowercase'),
+            }
+          )
+        );
+      }
+    } finally {
+      setIsButtonLoading(false);
+    }
+  };
+
+  const [selectedTrigger] =
+    Form.useWatch<CreateEventSubscription['resources']>(['resources'], form) ??
+    [];
+
+  const supportedFilters = useMemo(
+    () =>
+      entityFunctions.find((resource) => resource.name === selectedTrigger)
+        ?.supportedFilters,
+    [entityFunctions, selectedTrigger]
+  );
+
+  const shouldShowFiltersSection = useMemo(
+    () => (selectedTrigger ? !isEmpty(supportedFilters) : true),
+    [selectedTrigger, supportedFilters]
+  );
+
+  if (loadingCount) {
+    return <Loader />;
```

Comment:
I think this will show the loader in case the loadingCount is negative, so can we add a check for if l`oadingCount>0` then only render the loader?